docs/rfcs/2023-03-29-file-external-table/rfc.md

```diff
@@ -0,0 +1,86 @@
+---
+Feature Name: "File external table"
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/1041
+Date: 2023-03-08
+Author: "Xu Wenkang <wenymedia@gmail.com>"
+---
+
+File external table
+
+---
+
+# Summary
+
+Allows users to perform SQL queries on files
+
+# Motivation
+
+User data may already exist in other storages, i.g., file systems/s3, etc. in CSV, parquet, JSON format, etc. We can provide users the ability to perform SQL queries on these files.
+
+
+# Details
```

Comment:
Details lack an user interface section: how users are expected to use external tables?
1. Does external table needs to be created like MySQL or it can be specified per query like [clickhouse](https://clickhouse.com/docs/en/engines/table-engines/special/external-data)?
2. The life time of external table: If users have to manually create an external table, does it make sense to destroy the table automatically along with the session? Or the external table must be dropped manually?

Reply:
1. Maybe we can provide different scope external tables
The clickhouse has [something](https://clickhouse.com/docs/en/sql-reference/table-functions/s3) like:

```SQL
SELECT *
FROM s3(
   'https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv'
)
LIMIT 5;
```

It shares the low-level implements.

2. It's by our design. Andy said the different systems do different things:) The clickhouse treats the external tables as temporary tables, and they will be removed after the session is closed or disconnected. Maybe we can take a simple survey or make it configurable(but I don't think the configurable is a good idea. We still need the thinking ourselves; maybe we can infer from the typical use cases).

Reply:
Well, if the "external table" is durable, it has to survive or out live the session.